File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"

std::string raindrops::convert(int num)
{
        std::string result = "";

        if (num % 3 == 0) {
                result += "Pling"; 
        }

        if (num % 5 == 0) {
                result += "Plang"; 
        }

        if (num % 7 == 0) {
                result += "Plong"; 
        }

        if (result == "") {
                result = std::to_string(num); 
        }

        return result;
}

File: javascript/raindrops/raindrops.js
const RAINDROPS = new Map([[3, "Pling"], [5, "Plang"], [7, "Plong"]]);

export const convert = input => {
  let result = "";

  // Map.foreach gives value (word) first, then key (num)
  RAINDROPS.forEach((word, num) => {
    input % num === 0 ? (result += word) : null;
  });

  result === "" ? (result += input) : null;

  return result;
};
